src/Psl/Locale/Locale.php

```diff
@@ -791,6 +795,48 @@ enum Locale: string
     case Zulu = "zu";
     case ZuluSouthAfrica = "zu_ZA";
 
+    /**
+     * Retrieves the system's default locale from the PHP environment settings.
+     *
+     * This method returns the locale configured via `intl.default_locale`. Should the PHP environment lack a specific
+     * locale configuration or if the configured locale is unsupported, it defaults to `self::English` representing
+     * the English language.
+     *
+     * The choice of English as the fallback is motivated by its global comprehension, role as a base language in
+     * technology and international communication, predominance in technical documentation and support resources,
+     * and its historical precedence in the tech industry. These factors ensure the fallback locale is both broadly
+     * accessible and consistent with international standards.
+     *
+     * @return self The default locale as an enum instance, sourced from PHP settings or `self::English` as the fallback.
+     *
+     * @see https://www.php.net/manual/en/locale.getdefault.php
+     */
+    public static function default(): self
+    {
+        $full_locale = NativeLocale::getDefault();
+        if (!$full_locale) {
+            // Fallback to English if no locale is set or supported.
+            return self::English;
+        }
+
+        $language = NativeLocale::getPrimaryLanguage($full_locale);
+        $script = NativeLocale::getScript($full_locale);
+        $region = NativeLocale::getRegion($full_locale);
+
+        $locale = Str\lowercase($language);
+        if ($script) {
+            $locale .= '_' . Str\capitalize($script);
+        }
+
+        if ($region) {
+            $locale .= '_' . Str\uppercase($region);
+        }
+
+        // Attempt to match the system-configured locale with a supported enum instance,
+        // defaulting to English if a precise match is unavailable.
+        return self::tryFrom($locale) ?: self::English;
+    }
```

Comment:
wondering if we should do more attempts here, example, if the configured locale is `ar_Arab_TN` ( langauge = Arabic, script = Arabic, region = Tunisia ), we won't be able to match the locale with anything, as we do not have `ar_Arab_TN` in the enum, so we will fallback to English, however, we do have `ar_TN`, or even just `ar`, that we can fallback to instead ( note, the default script for `ar` language is `Arab`, so it is extremely unlikely that you would encounter `ar_Arab` or `en_Latn`, as those are the "default" scripts for the language, but, they are still valid ).

This would require us to create multiple combinations of `language+region` ( e.g. `ar_TN` ), `language+script` ( e.g. `zh_Hans` ), `language+script+region` ( e.g `sr_Latn` ) and just `langauge` ( e.g. `en` ) and try matching against them from the most verbose ( `lang+script+region` ) to the least verbose ( `lang` ).

personally i think what we have now is enough, and we can add this feature in the future.


wdyt?

Reply:
It might be sufficient indeed.


Maybe an easier approach instead of trying to combine many values, is to: try full, try language, fallback.
Something like:

```php
self::tryFrom($local) ?? self::tryFrom($language) : self::English;
```

That would at least translate into `Local::Arabic` `ar` then?

Reply:
A good idea indeed! we could improve it further later.